server/src/test/java/org/apache/druid/server/coordinator/duty/KillUnusedSegmentsTest.java

```diff
@@ -898,4 +924,25 @@ void deleteLastKillTaskId(final String dataSource)
       observedDatasourceToLastKillTaskId.remove(dataSource);
     }
   }
+
+  private void updateUsedStatusLastUpdated(DataSegment segment, DateTime lastUpdatedTime)
```

Comment:
I see this pattern repeat in a few other tests, so we can perhaps common out this functionality into a base class and refactor the test classes in a separate PR. Thoughts?

Reply:
I think the only other test doing this is `SqlSegmentsMetadataManagerTest`. So I guess we can leave this method as is for now.

What might be more useful though is adding an `update()` method in `TestDerbyConnector` which does most of the boiler plate stuff. There are similar updates being done in several other tests.

So essentially, move the method `SqlSegmentsMetadataManagerTest.executeUpdate()` to `TestDerbyConnector`.

Reply:
+1. A small variation of this update in `SQLMetadataConnectorTest.testAlterSegmentTableAddLastUsed()` with all the boilerplate.